File: shared/src/commonMain/kotlin/com/divya/data/models/Puja.kt
package com.divya.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Puja(
    val id: String,
    val temple: Temple? = null,
    val deity: Deity? = null,
    val name: LocalizedText,
    val type: String,
    val description: PujaDescription,
    val duration: Int,
    val pricing: PujaPricing,
    val displayPrice: CurrencyAmount? = null,
    val benefits: List<String> = emptyList(),
    val bestFor: List<String> = emptyList(),
    val requirements: List<String> = emptyList(),
    val isWaitlistOnly: Boolean = true,
    val waitlistCount: Int = 0,
    val estimatedWaitWeeks: Int = 4,
    val videoDelivered: Boolean = true,
    val videoNote: String? = null,
    val prasadAvailable: Boolean = false,
    val prasadNote: String = "Prasad delivery - Coming Soon 🙏",
    val order: Int = 0,
    val isActive: Boolean = true,
)

@Serializable
data class PujaDescription(
    val short: String? = null,
    val full: String? = null,
    val whatHappens: String? = null,
    val nriNote: String? = null,
)

@Serializable
data class PujaPricing(
    val usd: Double = 0.0,
    val gbp: Double = 0.0,
    val cad: Double = 0.0,
    val aud: Double = 0.0,
    val aed: Double = 0.0,
)

@Serializable
data class CurrencyAmount(
    val amount: Double,
    val currency: String,
)

File: backend/src/models/Puja.js
import mongoose from "mongoose";

const pujaSchema = new mongoose.Schema({
  temple: { type: mongoose.Schema.Types.ObjectId, ref: "Temple", required: true },
  deity: { type: mongoose.Schema.Types.ObjectId, ref: "Deity", required: true },
  name: {
    en: { type: String, required: true },
    ml: { type: String, required: true },
    sa: String
  },
  slug: { type: String, unique: true, sparse: true },
  type: {
    type: String,
    enum: ["abhishekam", "archana", "homa", "special_seva", "deeparadhana", "kalasha_puja"],
    required: true
  },
  description: {
    short: String,
    full: String,
    whatHappens: String,
    nriNote: String
  },
  duration: Number,
  pricing: {
    usd: Number,
    gbp: Number,
    cad: Number,
    aud: Number,
    aed: Number
  },
  benefits: [String],
  bestFor: [String],
  requirements: [String],
  isWaitlistOnly: { type: Boolean, default: true },
  waitlistCount: { type: Number, default: 0 },
  estimatedWaitWeeks: { type: Number, default: 4 },
  videoDelivered: { type: Boolean, default: true },
  videoNote: String,
  prasadAvailable: { type: Boolean, default: false },
  prasadNote: { type: String, default: "Prasad delivery - Coming Soon 🙏" },
  order: Number,
  isActive: { type: Boolean, default: true }
});

export const Puja = mongoose.model("Puja", pujaSchema);
